dodola/core.py

```diff
@@ -546,14 +547,15 @@ def apply_wet_day_frequency_correction(ds, process):
 
     if process == "pre":
         # includes very small values that are negative in CMIP6 output
-        ds_corrected = ds.where(
-            ds > threshold, np.random.uniform(low=low, high=threshold)
+        ds[variable] = ds[variable].where(
+            ds[variable] >= threshold,
+            np.random.uniform(low=low, high=threshold, size=ds[variable].shape),
```

Comment:
the issue is here - changing from `np.random.uniform(low, high)` to `np.random.uniform(low, high, size)` changes the returned type from pure python `float` to `ndarray[np.float64]`. xarray coerces the former to `ds[variable].dtype`, but defers to the higher precision data type of the latter